Sources/MongoKitten/AggregateStage.swift

```diff
@@ -36,9 +39,26 @@ public struct AggregateBuilderStage {
     }
     
     public static func project(_ projection: Projection) -> AggregateBuilderStage {
+        func valueUnwrapper(_ value: Primitive) -> WireVersion?{
+            switch value {
+            case let value as Int32:
+                if value == 0 { return .mongo3_4 } // indicates excluded fields
+                return nil
+            case let value as String:
+                if value == "$$REMOVE" { return .mongo3_6 } // indicates conditionally excluded fields
+                return nil
+            case let value as Document:
+                return valueUnwrapper(value)
+            default:
+                return nil
+            }
+        }
+        
+        let minimalVersionRequired: WireVersion? = projection.document.values.compactMap{ valueUnwrapper($0) }.min()
```

Comment:
Similarly to the keypaths, it's not necessary to create a block for `compactMap`, since we already have a fitting function that we want to pass into this.

```swift
projection.document.values.compactMap(valueUnwrapper)
```

Reply:
Secondly, we'll also want the highest version in this list. Not the lowest.